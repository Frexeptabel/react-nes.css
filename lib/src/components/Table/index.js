"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Table = function Table(_ref) {
  var bordered = _ref.bordered,
      centered = _ref.centered,
      dark = _ref.dark,
      props = _objectWithoutProperties(_ref, ["bordered", "centered", "dark"]);

  var className = (0, _classnames.default)("nes-table", {
    "is-bordered": bordered,
    "is-centered": centered,
    "is-dark": dark
  });
  return _react.default.createElement("table", _extends({
    className: className
  }, props), props.children);
};

Table.propTypes = {
  bordered: _propTypes.default.bool,
  centered: _propTypes.default.bool,
  dark: _propTypes.default.bool
};
var _default = Table;
exports.default = _default;