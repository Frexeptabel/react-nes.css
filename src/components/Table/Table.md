**Table example:**

Bordered & centered:

```js
<Table bordered centered>
  <thead>
    <tr>
      <th>Foo</th>
      <th>Foo</th>
      <th>Baz</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Tender crack Jennys tea cup ye hang the jib yo-ho-ho swab handsomely
        case shot bowsprit spirits.
      </td>
      <td>
        Tender crack Jennys tea cup ye hang the jib yo-ho-ho swab handsomely
        case shot bowsprit spirits.
      </td>
      <td>
        Tender crack Jennys tea cup ye hang the jib yo-ho-ho swab handsomely
        case shot bowsprit spirits.
      </td>
    </tr>
  </tbody>
</Table>
```

Bordered & centered & dark:

```js
<Table bordered centered dark>
  <thead>
    <tr>
      <th>Foo</th>
      <th>Foo</th>
      <th>Baz</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Tender crack Jennys tea cup ye hang the jib yo-ho-ho swab handsomely
        case shot bowsprit spirits.
      </td>
      <td>
        Tender crack Jennys tea cup ye hang the jib yo-ho-ho swab handsomely
        case shot bowsprit spirits.
      </td>
      <td>
        Tender crack Jennys tea cup ye hang the jib yo-ho-ho swab handsomely
        case shot bowsprit spirits.
      </td>
    </tr>
  </tbody>
</Table>
```
