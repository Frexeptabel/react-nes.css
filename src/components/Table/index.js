import React from "react";
import classNames from "classnames";
import PropTypes from "prop-types";

const Table = ({ bordered, centered, dark, ...props }) => {
  const className = classNames("nes-table", {
    "is-bordered": bordered,
    "is-centered": centered,
    "is-dark": dark
  });

  return (
    <table className={className} {...props}>
      {props.children}
    </table>
  );
};

Table.propTypes = {
  bordered: PropTypes.bool,
  centered: PropTypes.bool,
  dark: PropTypes.bool
};

export default Table;
