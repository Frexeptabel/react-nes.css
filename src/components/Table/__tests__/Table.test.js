import React from "react";
import Table from "../index";
import renderer from "react-test-renderer";

test("The Table should be rendered properly", () => {
  const component = renderer.create(<Table />);
  expect(component).toMatchSnapshot();
});

test("The Table should be rendered properly with props", () => {
  let component = renderer.create(<Table bordered />);
  expect(component).toMatchSnapshot();

  component = renderer.create(<Table centered />);
  expect(component).toMatchSnapshot();

  component = renderer.create(<Table dark />);
  expect(component).toMatchSnapshot();

  component = renderer.create(<Table dark centered bordered />);
  expect(component).toMatchSnapshot();
});
