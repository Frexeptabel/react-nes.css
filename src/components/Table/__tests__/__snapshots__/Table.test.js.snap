// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Table should be rendered properly 1`] = `
<table
  className="nes-table"
/>
`;

exports[`The Table should be rendered properly with props 1`] = `
<table
  className="nes-table is-bordered"
/>
`;

exports[`The Table should be rendered properly with props 2`] = `
<table
  className="nes-table is-centered"
/>
`;

exports[`The Table should be rendered properly with props 3`] = `
<table
  className="nes-table is-dark"
/>
`;

exports[`The Table should be rendered properly with props 4`] = `
<table
  className="nes-table is-bordered is-centered is-dark"
/>
`;
